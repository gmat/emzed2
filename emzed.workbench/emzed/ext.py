import pkg_resources

for ep in pkg_resources.iter_entry_points("emzed_plugin", name="package"):
    print ep
    package = ep.load()
    exec "%s=package" % package.__name__
    del package
    del ep
del pkg_resources

