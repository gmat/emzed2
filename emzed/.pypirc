[distutils]
index-servers = 
  local

[local]
repository: http://127.0.0.1:8080
username:uschmitt
password:pillepalle

