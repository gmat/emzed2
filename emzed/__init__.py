# encoding:latin-1

from version import version as __version__


