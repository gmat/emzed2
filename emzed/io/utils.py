

def _prepare_path(path, extensions):

    import sys
    if isinstance(path, unicode):
        path = path.encode(sys.getfilesystemencoding())
    elif path is None:
        from .. import gui
        path = gui.askForSingleFile(extensions=extensions)
    return path
