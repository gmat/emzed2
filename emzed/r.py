from core.r_connect.r_executor import RInterpreter
