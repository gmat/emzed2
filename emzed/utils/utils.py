import emzed.core.data_types

def toTable(colName, iterable,  format_="", type_=None, title="", meta=None):
    return emzed.core.data_types.Table.toTable(colName, iterable, format_, type_, title, meta)

toTable.__doc__ = emzed.core.data_types.Table.toTable.__doc__

def formula(mf):
    """
    Creates formula object which allows addition and subtraction:

    .. pycon::

       import ms   !onlyoutput
       mf1 = ms.formula("H2O")
       mf2 = ms.formula("NaOH")
       mf3 = mf1 + mf2
       print str(mf3)
       print str(mf3 - mf1)

    Mass calculation is supported too:

    .. pycon::

       print mf1.mass()

    If you need some internal representation, you can get a dictionary.
    Keys are pairs of *(symbol, massnumber)*, where *massnumber = None*
    refers to the lowest massnumber. Values of the dictionary are counts:

    .. pycon::

       print mf1.asDict()
       mixed = ms.formula("[13]C2[14]C3")
       print mixed.asDict()


    """
    from emzed.core.chemistry import MolecularFormula
    return MolecularFormula(mf)

def addmf(formula0, *formulas):
    """
    Combines molecular formulas by addition and subtraction:

    .. pycon::

       import ms !onlyoutput
       print ms.addmf("H2O", "COOH")
       print ms.addmf("H2O", "COOH", "NaCl")

    A leading minus sign subtracts the formula following this sign:

    .. pycon::

       print ms.addmf("H2O2", "-H2O")
       print ms.addmf("H2O", "COOH", "NaCl", "-H2O2")
       print ms.addmf("(CH2)7COOH", "-C7")

    """

    mf0 = formula(formula0)
    for f in formulas:
        if f.startswith("-"):
            mf0 -= formula(f[1:])
        else:
            mf0 += formula(f)
    return str(mf0)


###


def openInBrowser(urlPath):
    """
    opens *urlPath* in browser, eg:

    .. pycon::
        ms.openInBrowser("http://emzed.biol.ethz.ch") !noexec

    """
    from PyQt4.QtGui import QDesktopServices
    from PyQt4.QtCore import QUrl
    import os.path

    url = QUrl(urlPath)
    scheme = url.scheme()
    if scheme not in ["http", "ftp", "mailto"]:
        # C:/ or something simiar:
        if os.path.splitdrive(urlPath)[0] != "":
            url = QUrl("file:///"+urlPath)
    ok = QDesktopServices.openUrl(url)
    if not ok:
        raise Exception("could not open '%s'" % url.toString())


def _recalculateMzPeakFor(postfix):
    def calculator(table, row, name, postfix=postfix):

        mzmin = table.get(row, "mzmin"+postfix)
        mzmax = table.get(row, "mzmax"+postfix)
        rtmin = table.get(row, "rtmin"+postfix)
        rtmax = table.get(row, "rtmax"+postfix)
        pm    = table.get(row, "peakmap"+postfix)
        mz = pm.representingMzPeak(mzmin, mzmax, rtmin, rtmax)
        return mz if mz is not None else (mzmin+mzmax)/2.0
    return calculator

def _hasRangeColumns(table, postfix):
    return all([table.hasColumn(n + postfix) for n in ["rtmin", "rtmax",
                                                 "mzmin", "mzmax", "peakmap"]])

def recalculateMzPeaks(table):
    #TODO: tests !
    """Adds mz value for peaks not detected with centwaves algorithm based on
       rt and mz window: needed are columns mzmin, mzmax, rtmin, rtmax and
       peakmap mz, postfixes are automaticaly taken into account"""
    postfixes = [ "" ] + [ "__%d" % i for i in range(len(table._colNames))]
    for postfix in postfixes:
        if _hasRangeColumns(table, postfix):
            mz_col = "mz" + postfix
            if table.hasColumn(mz_col):
                table.replaceColumn(mz_col, _recalculateMzPeakFor(postfix),
                                    format="%.5f", type_=float)
            else:
                table.addColumn(mz_col, _recalculateMzPeakFor(postfix),
                                format="%.5f", type_=float)

def startfile(path):
    import sys, os
    if sys.platform == "win32":
        path = path.replace("/", "\\") # needed for network paths
    return os.startfile(path)

