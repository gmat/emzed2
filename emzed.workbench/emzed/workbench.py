def run():
    print 42

def main():
    import os
    print os.system("pip install -i http://127.0.0.1:3142/root/dev/+simple emzed.core")
    import emzed.core
    print emzed.core



