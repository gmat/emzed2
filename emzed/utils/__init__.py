from utils import *
from isotope_calculator import isotopeDistributionTable, plotIsotopeDistribution
from formula_generator  import formulaTable

from load_utils import loadCSV, loadTable, loadPeakMap

from feature_detectors import runCentwave, runMatchedFilters, runMetaboFeatureFinder

from metlin import matchMetlin

from integration import integrate
