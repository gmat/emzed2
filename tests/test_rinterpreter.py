from emzed.r import RInterpreter, RError, RInterpreterFast
from emzed.utils import toTable


def test_native_types():

    ip = RInterpreter()
    _test_native_types(ip)

def test_native_types_fast():

    ip = RInterpreterFast()
    _test_native_types(ip)

def _test_native_types(ip):

    assert ip.execute("x <-3").x == 3
    assert ip.execute("x <-1.0").x == 1.0
    assert ip.execute("x <-'abc'").x == 'abc'

    ip.y = 42
    assert ip.execute("x <- y").x == 42

    ip.y = 1.0
    assert ip.execute("x <- y").x == 1.0

    ip.y = "abc"
    assert ip.execute("x <- y").x == "abc"


def test_tables(regtest):
    ip = RInterpreter()
    _test_tables(ip, regtest)

def test_tables_fast(regtest):
    ip = RInterpreterFast()
    _test_tables(ip, regtest)

def _test_tables(ip, regtest):
    t = toTable("a", [1, 2])

    # transfer Table tor R:
    ip.t = t

    # fetch Table from R
    assert ip.execute("s <- t").s.rows == t.rows

    # fetch pandas.DataFrame from R
    df = ip.get_raw("s")
    assert df.as_matrix().tolist() == [[1], [2]]

    df = ip.get_raw("mtcars")
    print >> regtest, df

    ip.ddf = df

    print >> regtest, ip.ddf
    print >> regtest, ip.mtcars


def test_r_error_pickling():
    import dill

    # loads failed because the old constructor or RError had no "default constructor"
    err = dill.loads(dill.dumps(RError("test")))
    assert err.value == "test"
