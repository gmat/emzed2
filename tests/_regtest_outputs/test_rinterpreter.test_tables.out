                      mpg  cyl   disp   hp  drat     wt   qsec  vs  am  gear  \
Mazda RX4            21.0    6  160.0  110  3.90  2.620  16.46   0   1     4   
Mazda RX4 Wag        21.0    6  160.0  110  3.90  2.875  17.02   0   1     4   
Datsun 710           22.8    4  108.0   93  3.85  2.320  18.61   1   1     4   
Hornet 4 Drive       21.4    6  258.0  110  3.08  3.215  19.44   1   0     3   
Hornet Sportabout    18.7    8  360.0  175  3.15  3.440  17.02   0   0     3   
Valiant              18.1    6  225.0  105  2.76  3.460  20.22   1   0     3   
Duster 360           14.3    8  360.0  245  3.21  3.570  15.84   0   0     3   
Merc 240D            24.4    4  146.7   62  3.69  3.190  20.00   1   0     4   
Merc 230             22.8    4  140.8   95  3.92  3.150  22.90   1   0     4   
Merc 280             19.2    6  167.6  123  3.92  3.440  18.30   1   0     4   
Merc 280C            17.8    6  167.6  123  3.92  3.440  18.90   1   0     4   
Merc 450SE           16.4    8  275.8  180  3.07  4.070  17.40   0   0     3   
Merc 450SL           17.3    8  275.8  180  3.07  3.730  17.60   0   0     3   
Merc 450SLC          15.2    8  275.8  180  3.07  3.780  18.00   0   0     3   
Cadillac Fleetwood   10.4    8  472.0  205  2.93  5.250  17.98   0   0     3   
Lincoln Continental  10.4    8  460.0  215  3.00  5.424  17.82   0   0     3   
Chrysler Imperial    14.7    8  440.0  230  3.23  5.345  17.42   0   0     3   
Fiat 128             32.4    4   78.7   66  4.08  2.200  19.47   1   1     4   
Honda Civic          30.4    4   75.7   52  4.93  1.615  18.52   1   1     4   
Toyota Corolla       33.9    4   71.1   65  4.22  1.835  19.90   1   1     4   
Toyota Corona        21.5    4  120.1   97  3.70  2.465  20.01   1   0     3   
Dodge Challenger     15.5    8  318.0  150  2.76  3.520  16.87   0   0     3   
AMC Javelin          15.2    8  304.0  150  3.15  3.435  17.30   0   0     3   
Camaro Z28           13.3    8  350.0  245  3.73  3.840  15.41   0   0     3   
Pontiac Firebird     19.2    8  400.0  175  3.08  3.845  17.05   0   0     3   
Fiat X1-9            27.3    4   79.0   66  4.08  1.935  18.90   1   1     4   
Porsche 914-2        26.0    4  120.3   91  4.43  2.140  16.70   0   1     5   
Lotus Europa         30.4    4   95.1  113  3.77  1.513  16.90   1   1     5   
Ford Pantera L       15.8    8  351.0  264  4.22  3.170  14.50   0   1     5   
Ferrari Dino         19.7    6  145.0  175  3.62  2.770  15.50   0   1     5   
Maserati Bora        15.0    8  301.0  335  3.54  3.570  14.60   0   1     5   
Volvo 142E           21.4    4  121.0  109  4.11  2.780  18.60   1   1     4   

                     carb  
Mazda RX4               4  
Mazda RX4 Wag           4  
Datsun 710              1  
Hornet 4 Drive          1  
Hornet Sportabout       2  
Valiant                 1  
Duster 360              4  
Merc 240D               2  
Merc 230                2  
Merc 280                4  
Merc 280C               4  
Merc 450SE              3  
Merc 450SL              3  
Merc 450SLC             3  
Cadillac Fleetwood      4  
Lincoln Continental     4  
Chrysler Imperial       4  
Fiat 128                1  
Honda Civic             2  
Toyota Corolla          1  
Toyota Corona           1  
Dodge Challenger        2  
AMC Javelin             2  
Camaro Z28              4  
Pontiac Firebird        2  
Fiat X1-9               1  
Porsche 914-2           2  
Lotus Europa            2  
Ford Pantera L          4  
Ferrari Dino            6  
Maserati Bora           8  
Volvo 142E              2  
mpg       cyl      disp       hp       drat     wt       qsec      vs       am       gear     carb    
float     int      float      int      float    float    float     int      int      int      int     
------    ------   ------     ------   ------   ------   ------    ------   ------   ------   ------  
21.000000 6        160.000000 110      3.900000 2.620000 16.460000 0        1        4        4       
21.000000 6        160.000000 110      3.900000 2.875000 17.020000 0        1        4        4       
22.800000 4        108.000000 93       3.850000 2.320000 18.610000 1        1        4        1       
21.400000 6        258.000000 110      3.080000 3.215000 19.440000 1        0        3        1       
18.700000 8        360.000000 175      3.150000 3.440000 17.020000 0        0        3        2       
18.100000 6        225.000000 105      2.760000 3.460000 20.220000 1        0        3        1       
14.300000 8        360.000000 245      3.210000 3.570000 15.840000 0        0        3        4       
24.400000 4        146.700000 62       3.690000 3.190000 20.000000 1        0        4        2       
22.800000 4        140.800000 95       3.920000 3.150000 22.900000 1        0        4        2       
19.200000 6        167.600000 123      3.920000 3.440000 18.300000 1        0        4        4       
17.800000 6        167.600000 123      3.920000 3.440000 18.900000 1        0        4        4       
16.400000 8        275.800000 180      3.070000 4.070000 17.400000 0        0        3        3       
... (7 rows) ...
33.900000 4        71.100000  65       4.220000 1.835000 19.900000 1        1        4        1       
21.500000 4        120.100000 97       3.700000 2.465000 20.010000 1        0        3        1       
15.500000 8        318.000000 150      2.760000 3.520000 16.870000 0        0        3        2       
15.200000 8        304.000000 150      3.150000 3.435000 17.300000 0        0        3        2       
13.300000 8        350.000000 245      3.730000 3.840000 15.410000 0        0        3        4       
19.200000 8        400.000000 175      3.080000 3.845000 17.050000 0        0        3        2       
27.300000 4        79.000000  66       4.080000 1.935000 18.900000 1        1        4        1       
26.000000 4        120.300000 91       4.430000 2.140000 16.700000 0        1        5        2       
30.400000 4        95.100000  113      3.770000 1.513000 16.900000 1        1        5        2       
15.800000 8        351.000000 264      4.220000 3.170000 14.500000 0        1        5        4       
19.700000 6        145.000000 175      3.620000 2.770000 15.500000 0        1        5        6       
15.000000 8        301.000000 335      3.540000 3.570000 14.600000 0        1        5        8       
21.400000 4        121.000000 109      4.110000 2.780000 18.600000 1        1        4        2       

mpg       cyl      disp       hp       drat     wt       qsec      vs       am       gear     carb    
float     int      float      int      float    float    float     int      int      int      int     
------    ------   ------     ------   ------   ------   ------    ------   ------   ------   ------  
21.000000 6        160.000000 110      3.900000 2.620000 16.460000 0        1        4        4       
21.000000 6        160.000000 110      3.900000 2.875000 17.020000 0        1        4        4       
22.800000 4        108.000000 93       3.850000 2.320000 18.610000 1        1        4        1       
21.400000 6        258.000000 110      3.080000 3.215000 19.440000 1        0        3        1       
18.700000 8        360.000000 175      3.150000 3.440000 17.020000 0        0        3        2       
18.100000 6        225.000000 105      2.760000 3.460000 20.220000 1        0        3        1       
14.300000 8        360.000000 245      3.210000 3.570000 15.840000 0        0        3        4       
24.400000 4        146.700000 62       3.690000 3.190000 20.000000 1        0        4        2       
22.800000 4        140.800000 95       3.920000 3.150000 22.900000 1        0        4        2       
19.200000 6        167.600000 123      3.920000 3.440000 18.300000 1        0        4        4       
17.800000 6        167.600000 123      3.920000 3.440000 18.900000 1        0        4        4       
16.400000 8        275.800000 180      3.070000 4.070000 17.400000 0        0        3        3       
... (7 rows) ...
33.900000 4        71.100000  65       4.220000 1.835000 19.900000 1        1        4        1       
21.500000 4        120.100000 97       3.700000 2.465000 20.010000 1        0        3        1       
15.500000 8        318.000000 150      2.760000 3.520000 16.870000 0        0        3        2       
15.200000 8        304.000000 150      3.150000 3.435000 17.300000 0        0        3        2       
13.300000 8        350.000000 245      3.730000 3.840000 15.410000 0        0        3        4       
19.200000 8        400.000000 175      3.080000 3.845000 17.050000 0        0        3        2       
27.300000 4        79.000000  66       4.080000 1.935000 18.900000 1        1        4        1       
26.000000 4        120.300000 91       4.430000 2.140000 16.700000 0        1        5        2       
30.400000 4        95.100000  113      3.770000 1.513000 16.900000 1        1        5        2       
15.800000 8        351.000000 264      4.220000 3.170000 14.500000 0        1        5        4       
19.700000 6        145.000000 175      3.620000 2.770000 15.500000 0        1        5        6       
15.000000 8        301.000000 335      3.540000 3.570000 14.600000 0        1        5        8       
21.400000 4        121.000000 109      4.110000 2.780000 18.600000 1        1        4        2       

